File: day4.py
with open('day4.txt', 'r') as idFile:
    rawIDList = list(idFile)
    idList = [id.strip().split(',') for id in rawIDList]

pairsCount = 0
for id in idList:
    if (float(id[1].split('-')[1]) >= float(id[0].split('-')[1])) and (float(id[1].split('-')[0]) <= float(id[0].split('-')[0])):
        pairsCount += 1
    elif (float(id[0].split('-')[1]) >= float(id[1].split('-')[1])) and (float(id[0].split('-')[0]) <= float(id[1].split('-')[0])):
        pairsCount += 1

print(idList[0][1].split('-')[1])

print(f'The number of pairs fully contained in another is {pairsCount}')

###################-Part-2#################################

idSplit = []
for id in idList:
    idSplit += [[id[0].split('-'), id[1].split('-')]]

overlap = 0

for id in idSplit:
    first = int(id[0][0])
    second = int(id[0][1])
    third = int(id[1][0])
    fourth = int(id[1][1])
    if (((first > third) and (first > fourth)) and ((second > third) and (second > fourth))) or (((third > first) and (third > second)) and ((fourth > first) and (fourth > second))):
        continue
    else:
        overlap += 1

print(f'The numver of pairs that overlap are {overlap}')

File: day4/day4.js
const fs = require('fs');
const rawIDStr = fs.readFileSync('day4.txt').toString();

let rawIDList = rawIDStr.split('\n')

for (i = 0; i < rawIDList.length; i++) {
    rawIDList[i] = rawIDList[i].split(',')
    rawIDList[i][0] = rawIDList[i][0].split('-')
    rawIDList[i][1] = rawIDList[i][1].split('-')
    rawIDList[i][0][0] = parseInt(rawIDList[i][0][0])
    rawIDList[i][0][1] = parseInt(rawIDList[i][0][1])
    rawIDList[i][1][0] = parseInt(rawIDList[i][1][0])
    rawIDList[i][1][1] = parseInt(rawIDList[i][1][1])
}

let fullyContain = 0

for (i = 0; i < rawIDList.length; i++) {
    if ((rawIDList[i][1][1] >= rawIDList[i][0][1]) && (rawIDList[i][1][0] <= rawIDList[i][0][0])) {
        fullyContain += 1
    }
    else if ((rawIDList[i][0][1] >= rawIDList[i][1][1]) && (rawIDList[i][0][0] <= rawIDList[i][1][0])) {
        fullyContain += 1
    }
}

console.log(`The number of pairs fully contained in other is ${fullyContain}`)